src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ConstructedEETypeNode.cs

```diff
@@ -29,7 +29,43 @@ public override bool InterestingForDynamicDependencyAnalysis
         {
             get
             {
-                return _type.IsDefType && _type.HasGenericVirtualMethod();
+                if (_type.IsInterface)
+                    return _type.HasGenericVirtualMethods();
+
+                if (_type.IsDefType)
+                {
+                    // First, check if this type has any GVM that overrides a GVM on a parent type. If that's the case, this makes
+                    // the current type interesting for GVM analysis (i.e. instantiate its overriding GVMs for existing GVMDependenciesNodes
+                    // of the instantiated GVM on the parent types).
+                    foreach (var method in _type.GetAllMethods())
+                    {
+                        if (method.HasInstantiation && method.IsVirtual)
```

Comment:
True :) I'll leave that for a separate PR since this one already contains many changes.